chain-storage/src/tests.rs

```diff
@@ -774,14 +774,12 @@ fn permanent_store_prune_main_branch() {
         .prune_branch(&blocks.last().unwrap().id.serialize_as_vec())
         .unwrap();
 
-    for i in 0..=FLUSH_TO_BLOCK {
-        assert!(store
-            .block_exists(&blocks[i].id.serialize_as_vec())
-            .unwrap());
+    for block in blocks.iter().take(FLUSH_TO_BLOCK + 1) {
+        assert!(store.block_exists(&block.id.serialize_as_vec()).unwrap());
     }
 
-    for i in (FLUSH_TO_BLOCK + 1)..FLUSH_TO_BLOCK {
-        assert!(!store.block_exists(&blocks[i].serialize_as_vec()).unwrap());
+    for block in blocks.iter().skip(FLUSH_TO_BLOCK + 1) {
```

Comment:
here I'm actually guessing the original intent, as the existing code doesn't do anything.